api/src/main/java/com/velocitypowered/api/proxy/server/RegisteredServer.java

```diff
@@ -31,13 +34,49 @@
    * Returns a list of all the players currently connected to this server on this proxy.
    *
    * @return the players on this proxy
+   * @deprecated use {@link RegisteredServerState#getPlayersConnected()} instead
    */
-  Collection<Player> getPlayersConnected();
+  @Deprecated
+  default Collection<Player> getPlayersConnected() {
+    return (Collection<Player>) getState().getPlayersConnected();
+  }
 
   /**
    * Attempts to ping the remote server and return the server list ping result.
    *
    * @return the server ping result from the server
    */
   CompletableFuture<ServerPing> ping();
+
+  /**
+   * Returns the current state of this server on the proxy.
+   *
+   * @return the server state
+   */
+  RegisteredServerState getState();
+
+  /**
+   * Returns whether the server is registered in the proxy's server list. (i.e., in the config)
+   *
+   * <p>If this is {@code false}, then {@link com.velocitypowered.api.proxy.ProxyServer#getServer(String)}
+   * might return an empty {@link java.util.Optional}.
+   * </p>
+   *
+   * @return whether the server is in the config
+   */
+  boolean isRegistered();
```

Comment:
This seems like an implementation detail. A `RegisteredServer` surely is registered. We could add a note to `ProxyServer#getServer` documenting that it may return an empty optional depending on the registration means.

Reply:
True, though ultimately we'd want to rename the class. The main premise of the change is to allow servers to not be "registered" entirely, and just work as abstractions over a server connection.